actionpack/test/controller/redirect_test.rb

```diff
@@ -620,6 +641,180 @@ def test_redirect_to_external_with_rescue
+  end
+
+  def test_redirect_to_relative_url_without_leading_slash_with_raise
+    old_config = ActionController::Base.relative_redirects_without_leading_slash
+    ActionController::Base.relative_redirects_without_leading_slash = :raise
+
+    error = assert_raise(ActionController::Redirecting::UnsafeRedirectError) do
+      get :redirect_to_relative_url_without_leading_slash
+    end
+
+    assert_equal 'Relative URL redirect without leading slash detected: "exmaple.com"', error.message
+  ensure
+    ActionController::Base.relative_redirects_without_leading_slash = old_config
+  end
+
+  def test_redirect_to_relative_url_starting_with_an_at_with_raise
+    old_config = ActionController::Base.relative_redirects_without_leading_slash
+    ActionController::Base.relative_redirects_without_leading_slash = :raise
+
+    error = assert_raise(ActionController::Redirecting::UnsafeRedirectError) do
+      get :redirect_to_relative_url_starting_with_an_at
+    end
+
+    assert_equal 'Relative URL redirect without leading slash detected: "@example.com"', error.message
+  ensure
+    ActionController::Base.relative_redirects_without_leading_slash = old_config
+  end
+
+  def test_redirect_to_absolute_url_does_not_log
+    old_config = ActionController::Base.relative_redirects_without_leading_slash
+    ActionController::Base.relative_redirects_without_leading_slash = :log
+
+    logger = ActiveSupport::LogSubscriber::TestHelper::MockLogger.new
+    old_logger = ActionController::Base.logger
+    ActionController::Base.logger = logger
+
+    get :redirect_to_url
+    assert_response :redirect
+    assert_equal "http://www.rubyonrails.org/", redirect_to_url
+    assert_empty logger.logged(:warn)
+
+    get :relative_url_redirect_with_status
+    assert_response :redirect
+    assert_equal "http://test.host/things/stuff", redirect_to_url
+    assert_empty logger.logged(:warn)
+  ensure
+    ActionController::Base.logger = old_logger
+    ActionController::Base.relative_redirects_without_leading_slash = old_config
+  end
+
+  def test_redirect_to_absolute_url_does_not_notify
+    old_config = ActionController::Base.relative_redirects_without_leading_slash
+    ActionController::Base.relative_redirects_without_leading_slash = :notify
+
+    events = []
+    ActiveSupport::Notifications.subscribe("unsafe_redirect.action_controller") do |*args|
+      events << ActiveSupport::Notifications::Event.new(*args)
+    end
+
+    get :redirect_to_url
+    assert_response :redirect
+    assert_equal "http://www.rubyonrails.org/", redirect_to_url
+    assert_empty events
+
+    get :relative_url_redirect_with_status
+    assert_response :redirect
+    assert_equal "http://test.host/things/stuff", redirect_to_url
+    assert_empty events
+  ensure
+    ActiveSupport::Notifications.unsubscribe("unsafe_redirect.action_controller")
+    ActionController::Base.relative_redirects_without_leading_slash = old_config
+  end
+
+  def test_redirect_to_absolute_url_does_not_raise
```

Comment:
I am not sure we want the counter tests for log AND raise but I added them for now to make sure/prove that it does not break flows we expect to work. Happy to remove/update them though

Reply:
I think it's fine.  We can always delete them later.